lightningd/peer_control.c

```diff
@@ -2256,6 +2257,7 @@ struct htlc_in_map *load_channels_from_wallet(struct lightningd *ld)
 							       channel,
 							       ld->htlcs_out,
 							       unconnected_htlcs_in)) {
+				tal_free(unconnected_htlcs_in);
```

Comment:
This is unlikely to ever get hit, and if it does, we're not going to get a memleak since we call `fatal` which kills us immediately.